books/workshops/references/xdoc-generation.lisp

```diff
@@ -0,0 +1,154 @@
+         (year (get-field-value "year" fields))
+         (pages (get-field-value "pages" fields))
+         (publisher (get-field-value "publisher" fields))
+         (booktitle (get-field-value "booktitle" fields))
+         (month (get-field-value "month" fields))
+         (isbn (get-field-value "isbn" fields))
+         (doi (get-field-value "doi" fields)))
+
+    (concatenate 'string
+                 "<li><b>" (format-title title) "</b>, "
+                 (format-author-names author)
+                 (if booktitle
+                     (concatenate 'string ", in <i>" (strip-brackets booktitle) "</i>")
+                   "")
+                 (if publisher
+                     (concatenate 'string ", " (strip-brackets publisher))
+                   "")
+                 (if year
+                     (concatenate 'string ", " (strip-brackets year))
+                   "")
+                 (if month
+                     (concatenate 'string ", " (strip-brackets month))
+                   "")
+                 (if pages
+                     (concatenate 'string ", " (format-pages pages))
+                   "")
+                 (if isbn
+                     (concatenate 'string ", ISBN: " (strip-brackets isbn))
+                   "")
+                 (if doi
+                     (concatenate 'string ", DOI: " (strip-brackets doi))
+                   "")
+                 ". [" (format-entry-type entry-type) "]"
+                 "</li>")))
+
+(defun generate-bibtex-entries-html (entries)
+  "Generate HTML list items for all BibTeX entries"
+  (if (endp entries)
+      ""
+    (let ((entry-pair (car entries)))
+      (if (and (consp entry-pair)
+               (stringp (car entry-pair))
+               (alistp (cdr entry-pair)))
+          (concatenate 'string
+                       (generate-bibtex-entry-html (cdr entry-pair))
+                       (generate-bibtex-entries-html (cdr entries)))
+        (generate-bibtex-entries-html (cdr entries))))))
+
+
+(defun generate-workshop-documentation-xdoc (entries)
+  "Generate single XDOC documentation for all workshop entries"
+  (let ((entry-count (len entries))
+        (entries-html (generate-bibtex-entries-html entries)))
+    `(defxdoc acl2-workshops
+       :parents (pubs-papers)
+       :short "ACL2 Workshop Papers and Publications"
+       :long ,(concatenate 'string
+                           "<p>This section contains documentation for papers and publications related to ACL2 workshops and conferences.</p>"
+                           "<p><b>Total number of entries:</b> "(if (natp entry-count)
+                                                                    (coerce (explode-atom entry-count 10) 'string)
+                                                                  "unkown")
+                           "</p>"
+                           "<ul>"
+                           entries-html
+                           "</ul>"))))
+
+(set-state-ok t)
```

Comment:
Instead of setting this flag, I would just provide `(declare (xargs :stobjs state))` to the following function where it is needed.